pkg/controller/atlasdeployment/searchindex.go

```diff
@@ -77,15 +76,15 @@ func (sr *searchIndexReconciler) Reconcile(spec *akov2.SearchIndex, previous *st
 		}
 	}
 
-	return sr.reconcileInternal(stateInAKO, stateInAtlas)
+	return sr.reconcileInternal(name, stateInAKO, stateInAtlas)
 }
 
-func (sr *searchIndexReconciler) reconcileInternal(stateInAKO, stateInAtlas *searchindex.SearchIndex) workflow.Result {
+func (sr *searchIndexReconciler) reconcileInternal(indexName string, stateInAKO, stateInAtlas *searchindex.SearchIndex) workflow.Result {
 	sr.ctx.Log.Debugf("starting reconciliation for index '%s'", sr.indexName)
 	defer sr.ctx.Log.Debugf("finished reconciliation for index '%s'", sr.indexName)
 
-	emptyInAtlas := stateInAtlas == nil
-	emptyInAKO := stateInAKO == nil
+	inAtlas := stateInAtlas != nil
+	inSpec := stateInAKO != nil
 
```

Comment:
Why? But it's okay, I guess

Reply:
I'd say Spec is more accurate and avoids explaining an acronym. We use AKO too much. But I agree we are not consistent, even in this PR.